site/content/blog/2025-02-21-hipcheck-3.11.0-release.md

```diff
@@ -0,0 +1,198 @@
+
+---
+
+## User-facing Improvements
+
+### 📦&nbsp;&nbsp;Plugin Configuration Error Reporting
+
+#### `hc ready` Upgrade
+
+With Hipcheck 3.11.0, [hc ready][hc_ready] now expects a policy file and
+behaves like a `hc check` dry-run; it starts and configures plugins but does not
+proceed to analysis.
+
+The `hc ready` subcommand was first written before Hipcheck transitioned to a
+plugin architecture. At that time, we knew every possible analysis and their
+requirements (including additional configuration files, expected binaries and
+environment variables, etc.). Thus, `hc ready` could be hard-coded to check
+those requirements. In the new plugin-based context, the set of analyses that
+Hipcheck will run depends on the input policy file, which may involve
+third-party analysis plugins with unknown requirements.  Hipcheck's "readiness"
+is now directly dependent on the readiness of the plugins a user will employ in
+a given analysis; we are in turn defining a plugin's "readinesss" as whether it
+successfully completes the configuration step. We encourage plugin developers to
+use this step to do all necessary file parsing and environment checking, and not
+to report configuration success unless your criteria are met.
+
+#### Better Configuration Error Reporting
+
+With the above changes, `hc ready` now informs us if the configuration of a
+plugin (based on the supplied policy file) failed. To help users better
+understand and address configuration errors that may arise, we made the
+following additional changes. Firstly, for transparency Hipcheck now prints the
+name and version of the plugins that failed configuration. We also expanded the
+range of configuration error types that plugins can report, so that plugin
+authors can be more clear to users about what caused the failure. This change is
+reflected in the `ConfigurationStatus` enum, part of our gRPC protocol
+definition at `hipcheck-common/proto/hipcheck/v1/hipcheck.proto`. The new errors
+are as follows:
+
+```
+// The plugin encountered an internal error, probably due to incorrect assumptions.
+CONFIGURATION_STATUS_INTERNAL_ERROR = 5;
+
+// A necessary plugin input file was not found.
+CONFIGURATION_STATUS_FILE_NOT_FOUND = 6;
+
+// The plugin's input data could not be parsed correctly.
+CONFIGURATION_STATUS_PARSE_ERROR = 7;
+
+// An environment variable needed by the plugin was not set.
+CONFIGURATION_STATUS_ENV_VAR_NOT_SET = 8;
+
+// The plugin could not find or run a needed program.
+CONFIGURATION_STATUS_MISSING_PROGRAM = 9;
+```
+
+The suite of plugins maintained by the Hipcheck team has been updated to make
+use of these new error types where appropriate. Finally, using the `miette`
+crate we have updated our plugins that do file parsing to return high-fidelity
+parsing error messages that show where in the file the error arose and suggested
+fixes.
+
+### 📦&nbsp;&nbsp;JSON-formatted Analysis Reporting
+
+Users can select between having Hipcheck emit its analysis report in the shell
```

Comment:
"emit its analysis report in the shell" -> "emit its analysis report in a human-readable format"